File: java/plugins/solaceMessageProcessor/src/main/java/com/applicationsx/trademessageprocessor/TradeMessageBuilder.java
package com.applicationsx.trademessageprocessor;

import fixprocessor.TradeOuterClass;

public class TradeMessageBuilder {
    public static TradeOuterClass.Trade newOrder(String orderId, String symbol, int quantity, double price) {
        return TradeOuterClass.Trade.newBuilder()
                .setNewOrder(TradeOuterClass.NewOrder.newBuilder()
                        .setOrderId(orderId)
                        .setSymbol(symbol)
                        .setQuantity(quantity)
                        .setPrice(price)
                        .build())
                .build();
    }

    public static TradeOuterClass.Trade cancel(String orderId) {
        return TradeOuterClass.Trade.newBuilder()
                .setCancel(TradeOuterClass.Cancel.newBuilder()
                        .setOrderId(orderId)
                        .build())
                .build();
    }

    public static TradeOuterClass.Trade amend(String orderId, int newQuantity, double newPrice) {
        return TradeOuterClass.Trade.newBuilder()
                .setAmend(TradeOuterClass.Amend.newBuilder()
                        .setOrderId(orderId)
                        .setNewQuantity(newQuantity)
                        .setNewPrice(newPrice)
                        .build())
                .build();
    }
} 

File: cpp/plugins/solaceMessageProcessor/TradeMessageBuilder.cpp
#include "Trade.pb.h"

class TradeMessageBuilder {
public:
    static fixprocessor::Trade NewOrder(const std::string& order_id, const std::string& symbol, int quantity, double price) {
        fixprocessor::Trade trade;
        auto* new_order = trade.mutable_new_order();
        new_order->set_order_id(order_id);
        new_order->set_symbol(symbol);
        new_order->set_quantity(quantity);
        new_order->set_price(price);
        return trade;
    }

    static fixprocessor::Trade Cancel(const std::string& order_id) {
        fixprocessor::Trade trade;
        auto* cancel = trade.mutable_cancel();
        cancel->set_order_id(order_id);
        return trade;
    }

    static fixprocessor::Trade Amend(const std::string& order_id, int new_quantity, double new_price) {
        fixprocessor::Trade trade;
        auto* amend = trade.mutable_amend();
        amend->set_order_id(order_id);
        amend->set_new_quantity(new_quantity);
        amend->set_new_price(new_price);
        return trade;
    }
}; 